src/Features/CSharp/Portable/CodeRefactorings/InlineTemporary/InlineTemporaryCodeRefactoringProvider.cs

```diff
@@ -204,6 +212,10 @@ private async Task<Document> InlineTemporaryAsync(Document document, VariableDec
 
             var originalInitializerSymbolInfo = semanticModel.GetSymbolInfo(variableDeclarator.Initializer.Value, cancellationToken);
 
+            syntaxRoot = await updatedDocument.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var hasPossibleSideEffects = references.Count() > 1 &&
+                HasPossibleSideEffects(variableDeclarator.Initializer?.Value, syntaxRoot, references);
```

Comment:
i don't think we woudl allow inlining i the the declarator had a null initialier (i.e. nothing to initialize) so the `?.` isn't needed here.

also, doc this line and why we like it as a reasonable heuristic.